File: tools/sources/python/classes.py
"""Python 3 class declarations and features."""


class User:
    """Basic class with instance variables."""

    def __init__(self, name: str, email: str):
        self.name = name
        self.email = email
        self.active = True

    def greet(self):
        return f"Hello, {self.name}"


class Counter:
    """Class with class variables."""

    total_count = 0

    def __init__(self):
        self.count = 0
        Counter.total_count += 1

    def increment(self):
        self.count += 1


class Rectangle:
    """Class with instance methods."""

    def __init__(self, width: float, height: float):
        self.width = width
        self.height = height

    def area(self) -> float:
        return self.width * self.height

    def perimeter(self) -> float:
        return 2 * (self.width + self.height)


class Animal:
    """Base class for inheritance."""

    def __init__(self, name: str):
        self.name = name

    def speak(self):
        pass


class Dog(Animal):
    """Derived class."""

    def speak(self):
        return f"{self.name} says Woof!"

File: tools/sources/java/Classes.java
// Java 21 class declarations and features

public class Classes {

    // Basic class with fields
    public static class User {
        private String name;
        private String email;
        private int age;

        public User(String name, String email, int age) {
            this.name = name;
            this.email = email;
            this.age = age;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

    // Class with static members
    public static class Counter {
        private static int totalCount = 0;
        private int count;

        public Counter() {
            count = 0;
            totalCount++;
        }

        public static int getTotalCount() {
            return totalCount;
        }

        public void increment() {
            count++;
        }
    }

    // Inheritance
    public static abstract class Animal {
        protected String name;

        public Animal(String name) {
            this.name = name;
        }

        public abstract String speak();
    }

    public static class Dog extends Animal {
        public Dog(String name) {
            super(name);
        }

        @Override
        public String speak() {
            return name + " says Woof!";
        }
    }

    // Inner class
    public static class Outer {
        private int value = 10;

        public class Inner {
            public int getValue() {
                return value;
            }
        }
    }

    // Static nested class
    public static class Container {
        public static class Node {
            private int data;

            public Node(int data) {
                this.data = data;
            }
        }
    }
}
